continuationsPlugin/src/test/scala/continuations/CompilerFixtures.scala

```diff
@@ -174,25 +194,298 @@ trait CompilerFixtures { self: FunSuite =>
+      val c = summon[Context]
+      val suspend = Symbols.requiredClass("continuations.Suspend")
+      val intTpe = c.definitions.IntType
+      val ctxFunctionTpe = c
+        .definitions
+        .asContextFunctionType(c.definitions.FunctionOf(List(intTpe), intTpe, true, false))
+      ValDef(
+        Symbols
+          .newSymbol(
+            c.owner,
+            Names.termName("mySuspend"),
+            Flags.EmptyFlags,
+            ctxFunctionTpe
+          )
+          .asTerm,
+        Literal(Constant(1))
+      )
+    },
+    teardown = _ => ()
+  )
+
+  val zeroArityNonSuspendNonSuspendingDefDef = FunFixture[Context ?=> DefDef](
+    setup = _ => {
+      val c = summon[Context]
+      val suspend = Symbols.requiredClass("continuations.Suspend")
+      DefDef(
+        Symbols
+          .newSymbol(
+            c.owner,
+            Names.termName("mySuspend"),
+            Flags.EmptyFlags,
+            c.definitions.IntType)
+          .asTerm,
+        List(
+          List(),
+          List(Symbols.newSymbol(c.owner, Names.termName("s"), Flags.Param, suspend.info))),
+        c.definitions.IntType,
+        Literal(Constant(1))
+      )
+    },
+    teardown = _ => ()
+  )
+
+  val zeroAritySuspendSuspendingDefDef = FunFixture[Context ?=> DefDef](
+    setup = _ => {
+      val c = summon[Context]
+      val suspend = Symbols.requiredClass("continuations.Suspend")
+      val continuation = Symbols.requiredModule("continuations.Continuation")
+      val right = Symbols.requiredModule("scala.util.Right")
+
+      val intType = c.definitions.IntType
+
+      val usingSuspend =
+        Symbols.newSymbol(
+          c.owner,
+          Names.termName("x$1"),
+          Flags.union(Flags.GivenOrImplicit, Flags.Param),
+          suspend.info)
+
+      val anonFunc =
+        Symbols.newAnonFun(c.owner, continuation.companionClass.info)
+
+      val continuationVal = Symbols.newSymbol(
+        c.owner,
+        Names.termName("continuation"),
+        EmptyFlags,
+        continuation.companionClass.typeRef.appliedTo(intType)
+      )
+
+      val rightOne =
+        ref(right)
+          .select(Names.termName("apply"))
+          .appliedToTypes(List(c.definitions.ThrowableType, intType))
+          .appliedTo(Literal(Constant(1)))
+
+      val rhs =
+        Apply(
+          Apply(
+            TypeApply(
+              ref(continuation).select(Names.termName("suspendContinuation")),
+              List(TypeTree(intType))),
+            List(
+              Block(
+                Nil,
+                Block(
+                  List(DefDef(
+                    anonFunc,
+                    List(List(continuationVal)),
+                    c.definitions.UnitType,
+                    Block(
+                      Nil,
+                      ref(continuationVal).select(Names.termName("resume")).appliedTo(rightOne))
+                  )),
+                  Closure(Nil, ref(anonFunc), TypeTree(c.definitions.UnitType))
```

Comment:
btw there is another [Closure](https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/ast/tpd.scala#L115) which is supposed to create the whole [DefDef Block](https://github.com/47deg/TBD/pull/53/files#diff-d31a1b05e1eddc0b90913369d30ab6c242a20e2ee437a83f0f338652df2b6122R384) but I didn't manage to make it work with that one.

maybe you can give it a go, if it makes the code simpler

Reply:
I think it would. I'll give it a go.

Reply:
Yeah, couldn't get the tree to match the compiler output for the tree -- it misses the continuation named argument. Too bad, because it does simplify the code a bit.